help/destinations/catalog/email-marketing/sendgrid.md

```diff
@@ -0,0 +1,201 @@
+
+* Select the *Catalog* tab and search for *Sendgrid*. Then select *Activate Segments*.
+
+![](../../assets/catalog/email-marketing/sendgrid/02-catalog.jpg)
+
+* You will be shown a wizard where the first screen is to choose a destination. For setting up this connector, let us create a new destination by selecting *Configure new destination*.
+
+![](../../assets/catalog/email-marketing/sendgrid/03.jpg)
+
+* Select the *New Account* option and fill in the *Bearer Token* value which is the Sendgrid *API Key* value we had previously copied within [prerequisites](#prerequisites "Goto prerequisites").
+
+![](../../assets/catalog/email-marketing/sendgrid/04.jpg)
+
+* Upon selecting *Connect to destination*, you will be shown the next [step](#connection-parameters-parameters) to fill in additional information fields.
+
+### Connection parameters {#parameters}
+
+While [setting up](https://experienceleague.adobe.com/docs/experience-platform/destinations/ui/connect-destination.html?lang=en) this destination, you must provide the following information:
+
+*  **[!UICONTROL Name]**: The name by which you will recognize this destination in the future.
+*  **[!UICONTROL Description]**: An optional description that will help you identify this destination in the future.
+
+![](../../assets/catalog/email-marketing/sendgrid/05.jpg)
+
+![](../../assets/catalog/email-marketing/sendgrid/06.jpg)
+
+## Activate segments to this destination {#activate}
+
+Read [Activate profiles and segments to streaming segment export destinations](../../ui/activate/activate-segment-streaming-destinations.md) for instructions on activating audience segments to this destination. Steps specific to this destination are illustrated in detail below.
+
+![](../../assets/catalog/email-marketing/sendgrid/13.png)
+
+![](../../assets/catalog/email-marketing/sendgrid/14.jpg)
+
+> [!IMPORTANT]
+> The *Source* field attribute *"Email"* should be mapped to the *Target* field attribute "external_id" as shown in the subsequent step.
+
+![](../../assets/catalog/email-marketing/sendgrid/15.jpg)
+
+![](../../assets/catalog/email-marketing/sendgrid/16.jpg)
+
+* Similarly we map the remaining AEP fields and attributes to the Sendgrid attributes. For e.g. we are mapping the person object's first name atrribute below
+
+![](../../assets/catalog/email-marketing/sendgrid/17.jpg)
+
+![](../../assets/catalog/email-marketing/sendgrid/18.jpg)
+
+* A preview of some of the mapping's is as below
+
+![](../../assets/catalog/email-marketing/sendgrid/19.jpg)
+
+* The comprehensive list of the mappings that would be required to be setup is listed below.
+
+| Name | Field | Type | Description | Limits |
+|---|---|---|---|---|
+| contacts | - | Array of Objects | One or more contacts objects that you intend to upsert. The available fields for a contact, including the required email field are described below | maxItems: 30000 and minItems: 2 |
```

Comment:
I always appreciate a table, but I have a few questions here: 

1. what is the contacts field, why is it in a different column than all other fields? 
2. as far as I know, we don't allow exporting arrays of objects. Can you please check with the dev team that Array of Objects is correct?
3. Where are all the limits imported from? Is this from Sendgrid? Can we also link to the Sendgrid doc that specifies these limits? also, let's be more specific about the limits, for example `MaxLength: 100` can we edit to `MaxLength: 100 characters`?. the same goes for all limits documented in this column.
4. in the Description field, can we end all sentences with a fullstop punctuation, please? 
5. what is the `unique_name` field? Can we document this as well?
6. If custom_fields is not supported, why are we documenting it?

Reply:
1. The API accepts a contacts array where each contact is an object. Within the table thought the best way to represent this hierarchy / association is to split into multiple columns

Reply:
4. Actioned
6. Actioned

Reply:
6. Actioned

Reply:
3. Link updated, content updated.

Reply:
5. Removed

Reply:
2. Reference edited out